### DISCUSSION TITLE
Automatic register of mails and notifications

### DISCUSSION THREAD

**Discussion by juankOrtiz** [Xammie/mailbook#41] [Ideas]
Maybe instead of manually adding the mails and notifications there could be a way of auto-registering. I'm thinking something like the following steps:

1. Read the folders for mails and notifications
2. Check if there's something in each folder
3. If there are files, then loop trough each file and add them to mailbook. Maybe with a foreach loop?

```
foreach ($mails as $mail) {
    // Extract class from $mail, then:
    Mailbook::add(MailClass::class);
}
```

**Comment by Xammie** [comment]
This idea crossed my mind before. But I have not yet tried implementing it.

Most mails and notifications require specific parameters. For example your mail might need a user model and maybe some other model. We can create a new instance and pass it ofcourse (just like dependency injection), but I don't think this will be enough in most real world cases. I worry if this is not implemented right it will make the experience with mailbook confusing and unpredictable.

That being said I would certainly be open to suggestions or a PR that might solve this.

**Reply by juankOrtiz** [reply]
That make sense, I did not think about the specific parameters every type of mail would need.
Thanks again for your work on these awesome tool.
